src/mongocxx/exception/authentication.hpp

```diff
@@ -22,7 +22,10 @@ namespace mongocxx {
 MONGOCXX_INLINE_NAMESPACE_BEGIN
 namespace exception {
 
-class MONGOCXX_API authentication : public operation {};
+class MONGOCXX_API authentication : public base {
+   public:
+    authentication(bsoncxx::document::value raw_server_error);
```

Comment:
Having looked at several of these exception declarations, I'm pretty convinced that you can capitalize on constructor inheritance fully. Just using in the base constructors you want, in each derived exception, via the immediate super classes name.